packages/core/src/domain/conversion/bundle-modifications/modifications.ts

```diff
@@ -0,0 +1,106 @@
+import { Bundle, Resource } from "@medplum/fhirtypes";
+import { cloneDeep } from "lodash";
+import * as uuid from "uuid";
+import { isPatient } from "../../../external/fhir/shared";
+import { DOC_ID_EXTENSION_URL } from "../../../external/fhir/shared/extensions/doc-id-extension";
+
+export type FhirExtension = {
+  url: string;
+  valueString: string;
+};
+
+export type FhirConverterParams = {
+  patientId: string;
+  fileName: string;
+  unusedSegments: string | undefined;
+  invalidAccess: string | undefined;
+};
+
+/**
+ * Replaces the IDs (and all references to them) for all resources that have an extension for the source document.
+ * This is done so that identical resources originated from different sources would have different IDs.
+ */
+export function replaceIdsForResourcesWithDocExtension(
+  fhirBundle: Bundle<Resource>,
+  patientId: string
+): Bundle<Resource> {
+  const updatedBundle = cloneDeep(fhirBundle);
+  const stringsToReplace: { old: string; new: string }[] = [];
+  if (!updatedBundle.entry || updatedBundle.entry.length === 0) {
+    throw new Error(`Missing bundle entries`);
+  }
+  for (const bundleEntry of updatedBundle.entry) {
+    if (!bundleEntry.resource) throw new Error(`Missing resource`);
```

Comment:
NABD, and likely not to hit any case in real life, but since we're coding this, it's worth the convo: can't we just `continue`? Like, if there's an entry without resource, we don't need to break the whole flow, we just continue. The function's contract is still intact, and we minimize breaking the app - make it more resilient to diff states.

Reply:
BTW, that's what we're doing on `addExtensionToConversion`